crates/pipeline_manager/src/local_runner.rs

```diff
@@ -35,7 +35,16 @@ pub struct ProcessRunner {
 
 impl Drop for ProcessRunner {
     fn drop(&mut self) {
-        let _ = self.pipeline_process.as_mut().map(|p| p.kill());
+        self.kill_pipeline();
+    }
+}
+
+impl ProcessRunner {
+    fn kill_pipeline(&mut self) {
+        if let Some(mut p) = self.pipeline_process.take() {
+            let _ = p.kill();
+            let _ = p.wait();
```

Comment:
Is there a risk that `p.wait()` never returns?

Reply:
Good question.  In theory - yes. In practice, I think `kill` should be able to kill a process reliably. I tried using `try_wait`,  but calling it right after `kill` means that the process is likely still running, the `try_wait` call fails, and the child process ends up in a defunct state.